middleware_demo/roles/deploy_wildfly/tasks/main.yml

```diff
@@ -1,60 +1,26 @@
 ---
-- name: Add EPEL repository
-  yum_repository:
-    name: epel
-    description: EPEL YUM repo
-    file: epel_repos
-    baseurl: "{{ epel_baseurl }}"
-    gpgcheck: yes
-
-- name: Add GPG Key
-  rpm_key:
-    state: present
-    key: "{{ epel_rpm_key }}"
-
-- name: Upgrade all packages
-  yum:
-    name: '*'
-    state: latest
-
-- name: Install the required packages
-  yum:
-    name: "{{ item }}"
-    state: present
-  loop:
-    - xmlstarlet
-    - augeas
-    - bsdtar
-    - unzip
-    - java-11-openjdk-devel
-
-- name: Create jboss group
+- assert:
+    that:
+      - wildfly_user is defined
+      - wildfly_group is defined
+      - wildfly_uid is defined
+      - wildfly_gid is defined
+    quiet: true
+
+- name: "Create {{ wildfly_group }} group"
   group:
-    name: jboss
+    name: "{{ wildfly_group }}"
     state: present
-    gid: 1001
+    gid: "{{ wildfly_gid }}"
 
-- name: Create {{ jboss_home_directory }} directory
-  file:
-    path: "{{ jboss_home_directory }}"
-    state: directory
-    mode: '0755'
-
-- name: Create jboss user
+- name: "Create {{ wildfly_user }} user"
   user:
-    name: jboss
+    name: "{{ wildfly_user }}"
     comment: JBoss User
-    uid: 1001
-    group: jboss
+    uid: "{{ wildfly_uid }}"
+    group: "{{ wildfly_group }}"
     home: "{{ jboss_home_directory }}"
-
-- name: Update ownership of {{ jboss_home_directory }}
-  file:
-    path: "{{ jboss_home_directory }}"
-    state: directory
-    recurse: yes
-    owner: jboss
-    group: jboss
+    state: present
 
 # Add the WildFly distribution to /opt, and make wildfly the owner of the extracted tar content
 - name: Download and Add WildFly distribution
```

Comment:
@kahlai The download Wildfly part I did not change, but I think it needs a rehaul. First, I would not download and directly unzip into jboss_home. It's not idempotent and you need to download the archive every time you run the tasks, even if the content is already deployed. 
I would move the download part into the pre_tasks: part of the playbook and save the archive in something like /opt/apps or /opt/artefacts. Then, inside the role, I would check that wildfy_archive is defined and if so unarchive: the zipfile into jboss_home.

Reply:
Also, the series of commands in shell: are a bit cluncky. If you have that many commands to run, I would move that into a proper Bash script that I would then run. I also think Ansible has mechanisms to check the sha1sum, decompress tar file and ensure jboss:jboss is the owner of the file. 
We are doing just that in this [playbook](https://github.com/ansible-middleware/ansible_collections_jcliff/blob/master/playbook.yml#L37).